src/extlibs/hrvo/path.h

```diff
@@ -0,0 +1,72 @@
+#pragma once
+#include <memory>
+#include <vector>
+
+#include "extlibs/hrvo/path_point.h"
+#include "extlibs/hrvo/vector2.h"
+
+class PathPoint;
+
+/**
+ * An agent's path
+ */
+class Path
+{
+   public:
+    /**
+     * Default constructor for this path
+     */
+    explicit Path();
+
+    /**
+     * Constructor for this path
+     * @param path_points A vector that stores path points, which can be empty
+     * @param goal_radius The goal radius for this path
+     */
+    explicit Path(const std::vector<PathPoint> &path_points, float goal_radius);
+
+    /**
+     * Gets the next path point position in this path
+     * @return the position of the next path point
+     */
+    Vector2 getNextPathPointPosition();
```

Comment:
it seems overly verbose, i.e. what's wrong with nextPosition()? Also why can't it just return a PathPoint so we don't need two functions for point and speed

Reply:
I found it easier to understand while using two separate functions, though returning a PathPoint is also fine. I tried to make the functions self-descriptive but I agree it's verbose, will consider changing to the suggestion